service/history/replicatorQueueProcessor.go

```diff
@@ -471,6 +477,13 @@ func (p *replicatorQueueProcessorImpl) getTasks(
 		time.Duration(len(replicationTasks)),
 	)
 
+	if err := p.shard.UpdateClusterReplicationLevel(
+		sourceCluster,
+		lastReadTaskID,
```

Comment:
lastReadTaskID cannot be updated in this way. Imagine that this batch is totally lost before sending. Then poller will never see it again.
lastReadTaskID should be whatever value passed from the poller.

Reply:
Synced offline. the lastReadTaskID does not update and it is from the remote clusters